src/OpenDiffix.Core/Aggregator.fs

```diff
@@ -288,6 +288,115 @@ type private DiffixLowCount() =
       else
         Boolean(Anonymizer.isLowCount aggContext.AnonymizationParams anonContext state)
 
+type private DiffixSum() =
+  let nullState: Anonymizer.SumState = { Positive = null; Negative = null; IsReal = false }
+  let mutable state = nullState
+
+  let initialState length : Anonymizer.SumState =
+    { nullState with
+        Positive =
+          Array.init length (fun _ -> { AidContributions = Dictionary<AidHash, float>(); UnaccountedFor = 0.0 })
+        Negative =
+          Array.init length (fun _ -> { AidContributions = Dictionary<AidHash, float>(); UnaccountedFor = 0.0 })
+    }
+
+  let increaseUnaccountedFor valueIncrease i =
+    let absValueIncrease = abs (valueIncrease)
+
+    if valueIncrease > 0.0 then
+      state.Positive.[i].UnaccountedFor <- state.Positive.[i].UnaccountedFor + absValueIncrease
+    else if valueIncrease < 0.0 then
+      state.Negative.[i].UnaccountedFor <- state.Negative.[i].UnaccountedFor + absValueIncrease
+    else
+      state.Positive.[i].UnaccountedFor <- state.Positive.[i].UnaccountedFor + absValueIncrease
+      state.Negative.[i].UnaccountedFor <- state.Negative.[i].UnaccountedFor + absValueIncrease
+
+  let increaseSumContribution valueIncrease aidValue i =
+    let absValueIncrease = abs (valueIncrease)
+
+    if valueIncrease > 0.0 then
+      increaseContribution absValueIncrease aidValue state.Positive.[i].AidContributions
+    else if valueIncrease < 0.0 then
+      increaseContribution absValueIncrease aidValue state.Negative.[i].AidContributions
+    else
+      increaseContribution absValueIncrease aidValue state.Positive.[i].AidContributions
+      increaseContribution absValueIncrease aidValue state.Negative.[i].AidContributions
+
+
+  /// Increases contribution of all AID instances.
+  let increaseContributions valueIncrease (aidInstances: Value list) =
+    aidInstances
+    |> List.iteri (fun i aidValue ->
+      match aidValue with
+      // No AIDs, add to unaccounted value
+      | Null
+      | Value.List [] -> increaseUnaccountedFor valueIncrease i
+      // List of AIDs, distribute contribution evenly
+      | Value.List aidValues ->
+        let partialIncrease = valueIncrease / (aidValues |> List.length |> float)
+
+        aidValues
+        |> List.iter (fun aidValue -> increaseSumContribution partialIncrease aidValue i)
```

Comment:
Maybe we should get rid of this partial increase features completely? We are nowhere near a spec to handle it...

Reply:
If there's consensus, sure, but let's do it in a separate PR, since I think this is orthogonal, and the support for `sum` is already here.